### ISSUE TITLE
Atoum support

### ISSUE THREAD

**Issue by oxman** [padraic/humbug#136]
Can you add support of Atoum : http://docs.atoum.org/en/latest/

**Comment by aztech-dev** [comment]
@padraic need a volunteer ? My schedule's clearing up starting this week...

**Comment by padraic** [comment]
@aztech-dev Sure. If someone implements it, I'll be happy to add the support!

It looks like atoum has both code coverage and allows for filtering of what to run, so hopefully it's not a difficult task managing the optimisation. The output isn't TAP but hopefully is more predictable than PHPUnit's was.

We're hopefully going to move away from directly having testing frameworks as direct dependencies, so the preference would be to have it implemented using whatever is locally installed. Barring automatic detection, you can add options to describe what to use. e.g.

```json
{
    "timeout": 10,
    "source": {
        "directories": [
            "src"
        ]
    },
    "adapters": {
        "atoum"
    },
    "executables": {
        "atoum": "vendor/bin/atoum"
    }
}
```

**Comment by aztech-dev** [comment]
Ok I (think I) see where you wanna go :) I'll get started tomorrow evening.

**Comment by padraic** [comment]
@aztech-dev Yeah, @rollenes proposed it for PHPUnit and it seems to be a cleaner option than having own required dependency on every framework.

Unfortunately, time machines are in short supply ;). I'll take a look at doing the same for PHPUnit if he can't get to it.
